ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -554,6 +560,11 @@ private Optional<BlockHeader> getChainHeadBlockHeader() {
     return blockchain.getBlockHeader(blockchain.getChainHeadHash());
   }
 
+  private Optional<BlockHeader> getChainHeadBlockHeaderSafe() {
+    final MutableBlockchain blockchain = protocolContext.getBlockchain();
+    return blockchain.getBlockHeaderSafe(blockchain.getChainHeadHash());
+  }
+
```

Comment:
how about instead of adding here we add the safe call to the existing getChainHeadBlockHeader(), e.g.:

```
  private Optional<BlockHeader> getChainHeadBlockHeader() {
    final MutableBlockchain blockchain = protocolContext.getBlockchain();
    return blockchain.getBlockHeader(blockchain.getChainHeadHash())
        .or(() -> blockchain.getBlockHeaderSafe(blockchain.getChainHeadHash()));
  }
```

Reply:
See latest commits